feature-libs/checkout/base/core/facade/checkout-payment.service.spec.ts

```diff
@@ -0,0 +1,338 @@
+      const subscription = service
+        .getCardTypesState()
+        .subscribe((result) => (resultState = result));
+
+      tick(4200);
+
+      expect(resultState).toEqual({
+        loading: false,
+        error: <Error>mockJaloError,
+        data: undefined,
+      });
+      subscription.unsubscribe();
+    }));
+
+    xit(`should successfully backOff on Jalo error and recover after the 2nd attempt`, fakeAsync(() => {
+      spyOn(connector, 'getCardTypes').and.returnValues(
+        // first attempt
+        throwError(mockJaloError),
+        // second attempt
+        throwError(mockJaloError),
+        // third time the charm
+        of(mockCardTypes)
+      );
+
+      let resultState: QueryState<CardType[] | undefined> | undefined;
+      const subscription = service.getCardTypesState().subscribe((result) => {
+        console.log('result', result);
+        return (resultState = result);
+      });
+
+      // 1*1*300 = 300
+      tick(300);
+      expect(resultState).toEqual({
+        loading: true,
+        error: false,
+        data: undefined,
+      });
+
+      // 2*2*300 = 1200
+      tick(1200);
+      expect(resultState).toEqual({
+        loading: false,
+        error: false,
+        data: mockCardTypes,
+      });
+      subscription.unsubscribe();
+    }));
+  });
+
+  describe(`getCardTypes`, () => {
+    it(`should call facade's getCardTypesState()`, (done) => {
+      spyOn(service, 'getCardTypesState').and.returnValue(
+        of({
+          loading: false,
+          error: false,
+          data: mockCardTypes,
+        })
+      );
+
+      service
+        .getCardTypes()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(mockCardTypes);
+          done();
+        });
+    });
+
+    it(`should return an empty array if query's data is falsy`, (done) => {
+      spyOn(service, 'getCardTypesState').and.returnValue(
+        of({
+          loading: false,
+          error: false,
+          data: undefined,
+        })
+      );
+
+      service
+        .getCardTypes()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual([]);
+          done();
+        });
+    });
+  });
+
+  describe(`getPaymentDetailsState`, () => {
+    it(`should return the delivery modes`, (done) => {
+      spyOn(checkoutQuery, 'getCheckoutDetailsState').and.returnValue(
+        of(<QueryState<CheckoutState>>{
+          loading: false,
+          error: false,
+          data: {
+            paymentInfo: mockPaymentInfo,
+          },
+        })
+      );
+
+      service
+        .getPaymentDetailsState()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(<QueryState<PaymentDetails | undefined>>{
+            loading: false,
+            error: false,
+            data: mockPaymentInfo,
+          });
+          done();
+        });
+    });
+  });
+
+  describe(`createPaymentDetails`, () => {
```

Comment:
add 2 `it`

1 - should throw error when paymentDetails is undefined and expect no connector calls
2 - should not dispatch LoadUserPaymentMethods(xxx) when anonymous user

Reply:
1 - removed the checks, as it shouldn't be possible to pass a falsy value
2 - added